src/vm/assembly.cpp

```diff
@@ -1965,6 +1965,10 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs, BOOL waitForOtherThre
     IfFailThrow(hr);
     
     END_ENTRYPOINT_THROWS;
+
+    // Sync the latched exit code with the return value of main function.
+    SetLatchedExitCode(iRetVal);
```

Comment:
Is this really needed? 

It will incorrectly overwrite latched exit code for non-void returning functions.

https://github.com/mazong1123/coreclr/blob/0c46640307dab1d8359f4e3b856a4390d9a6389d/src/vm/assembly.cpp#L1854 should take care of it for non-void returning main.